pkg/custom-provider/provider.go

```diff
@@ -147,16 +147,18 @@ func (p *prometheusProvider) buildQuery(ctx context.Context, info provider.Custo
 	// TODO: use an actual context
 	queryResults, err := p.promClient.Query(ctx, pmodel.Now(), query)
 	if err != nil {
-		klog.Errorf("unable to fetch metrics from prometheus: %v", err)
+		klog.Errorf("unable to fetch metrics from metrics source: %s", err.Error())
```

Comment:
Is this a `*client.Error`? Can you log the other fields, e.g. status code, query, etc.

Reply:
it is a `*client.Error`

Calling `err.Error()` on the `*client.Error` object returns a string as follows
```
func (e *Error) Error() string {
	base := fmt.Sprintf("%s: %s", e.Type, e.ErrorMsg)
	if e.Query != "" {
		return fmt.Sprintf("%s for status code: %d, query: %s", base, e.StatusCode, e.Query)
	}
	return base
}
```
This handles missing query field (which shouldn't happen), and also prints out all the info of the query.